### ISSUE TITLE
model._function_kwargs is silently ignored

### ISSUE THREAD

**Issue by bersbersbers** [tensorflow/tensorflow#35783]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): see below
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows, Linux
- TensorFlow installed from (source or binary): pip
- TensorFlow version (use command below): 2.1.0
- Python version: 3.7.6

**Describe the current behavior**
In this example, no error is raised although session kwargs are not supported in eager mode. kwargs are simply silently ignored.
```
import tensorflow as tf
from tensorflow import keras

fetches = [lambda: whatever_I_write_here_is_ignored]

var = tf.Variable([[3.0]])
model = keras.models.Sequential([keras.layers.Dense(1, input_shape=(1,))])
model.compile(loss="mse", optimizer="adam")
model._function_kwargs = {"fetches": fetches, "should_fail": "ignored_as_well"}

model.fit([[7.0]], [[9.0]], epochs=2)
```

**Describe the expected behavior**
An error should be raised. I am aware that `model._function_kwargs` is not part of the public API, but `keras` (as opposed to `tf.keras`) does raise an error here:
```
import keras
import tensorflow as tf

fetches = [lambda: whatever_I_write_here_is_ignored]

var = tf.Variable([[3.0]])
model = keras.models.Sequential([keras.layers.Dense(1, input_shape=(1,))])
model.compile(loss="mse", optimizer="adam")
model._function_kwargs = {"fetches": fetches, "should_fail": "ignored_as_well"}

model.fit([[7.0]], [[9.0]], epochs=2)
```

outputs
```
Exception has occurred: ValueError
Session keyword arguments are not support during eager execution. You passed: {'fetches': [<function <lambda> at 0x7f94681be3b0>], 'should_fail': 'ignored_as_well'}
  File "/home/bersbersbers/.pyenv/versions/3.7.6/lib/python3.7/site-packages/tensorflow_core/python/keras/backend.py", line 3759, in function
```
Somewhat related: https://github.com/tensorflow/tensorflow/issues/34448

**Comment by gadagashwini** [comment]
I tried replicating issue with Tf 2.1 but facing different error.
Please find the [gist](https://colab.sandbox.google.com/gist/gadagashwini/b744dcb3fe10d5c97560dce02220af9e/untitled344.ipynb?authuser=1) and confirm. Thanks!

**Comment by bersbersbers** [comment]
@gadagashwini you are reproducing the issue just fine. The issue is that in the first example using `from tensorflow import keras` , `fetches = [lambda: whatever_I_write_here_is_ignored]` and `model._function_kwargs = {"fetches": fetches, "should_fail": "ignored_as_well"}` are silently ignored. This is the issue I am reporting.

I agree that the comparison `import keras` case behaves differently by issuing a different error, but that is not my point. My point is that `keras` is correct to throw an error, and that `tf.kers` is incorrect in not doing the same.

**Comment by ymodak** [comment]
This is fixed with latest tf-nightly '2.2.0-dev20200325'. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by bersbersbers** [comment]
I can still reproduce this issue on `tensorflow==2.2.0rc2`; this version was released to PyPI on March 28, but I don't know how this related to March 25's nightly that is supposedly fixed.

**Comment by bersbersbers** [comment]
Still an issue with `tensorflow==2.2.0rc3`, just in case anyone is wondering.

**Comment by Saduf2019** [comment]
@bersbersbers 
I ran the code on tf-nightly [2.4.0-dev20200819] and do not face any issues, please find the [gist here](https://colab.research.google.com/gist/Saduf2019/e0171158ebee39eb2d30adb7054af8bc/untitled380.ipynb).

**Comment by bersbersbers** [comment]
@Saduf2019 your code does not raise an error. That is the issue, you are reproducing it just fine. Please see https://github.com/tensorflow/tensorflow/issues/35783#issuecomment-573662235 for context.

**Comment by Saduf2019** [comment]
This issue is still persist on 2.4.0-dev20200819

**Comment by sachinprasadhs** [comment]
The issue still exists in TF nightly version(2.6.0-dev20210524), please find the gist [here](https://colab.research.google.com/gist/sachinprasadhs/122c36f5154898369075959bcc273e7c/35650.ipynb).

**Comment by bersbersbers** [comment]
I have come to the conclusion that this can be closed, though. The reference code (using standalone `keras`) does not throw an error, either. And the code using an undocumented API is from TF 2.1 and can be considered outdated.
